src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java

```diff
@@ -45,7 +45,15 @@
     private static final Map<String, Class<? extends AbstractJob>> nameToClassMap = new HashMap<>();
     @NotNull
     private static final Map<Class<? extends AbstractJob>, String> classToNameMap = new HashMap<>();
-    static
+
+    /*
+     * Suppressing Sonar Rule squid:S2390
+     * This rule does "Classes should not access static members of their own subclasses during initialization"
+     * But in this case the rule does not apply because
+     * We are only mapping classes and that is reasonable
+     */
+    @SuppressWarnings("squid:S2390")
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S2390' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)